unity_common/src/Plugin/Field/FieldFormatter/TaxonomyToFacetFormatter.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\unity_common\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\facets\Entity\Facet;
+
+/**
+ * Plugin implementation of the 'Taxonomy term to facet' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "Map taxonomy term to facet",
+ *   label = @Translation("Unity taxonomy term to facet formatter"),
+ *   field_types = {
+ *     "entity_reference"
+ *   },
+ * )
+ */
+class TaxonomyToFacetFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+        'search_page_url' => '',
+        'facets' => ''
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $elements = parent::settingsForm($form, $form_state);
+
+    // TODO: There is no form validator call for plugin settings forms,
+    // possibly look at using an ajax callback to validate CSS classes.
+    $facet_manager = \Drupal::service('facets.manager');
```

Comment:
Use DI to inject this service: https://www.drupal.org/docs/drupal-apis/services-and-dependency-injection